iphone/Maps/Categories/UIKitCategories.m

```diff
@@ -205,7 +205,7 @@ - (BOOL)openUrl:(NSString *)urlString inSafari:(BOOL)safari
     urlc.scheme = @"http";
 
   NSURL * url = urlc.URL;
-  if (safari)
+  if (externally && [UIApplication.sharedApplication canOpenURL:url])
```

Comment:
If externally is passed, that implies that a safari browser app will be opened. But this check works only if an app is installed, right?

Reply:
Yes. This is the `[UIApplication.sharedApplication canOpenURL:url]`'s responsibility to check the app is installed. In other case Safari will be used.

Reply:
So now the behavior is:
1. externally == YES: either an external app opens, or in-OM SafariViewController is opened.
2. externally = NO: in-OM SafariViewController is opened

Correct?

Would it be clearer to have the behavior below?
1. externally == YES: either an external app opens, or an external browser is opened
2. externally = NO: in-OM SafariViewController is opened

Maybe rename `externally` to something like `openInTheApp` (and reverse YES/NO)?

Reply:
> 1. externally == YES: open the link by opening the some other application and fully leaving the OM. If ios can't find the proper app it will open it in Safari app (external browser).

So `[UIApplication.sharedApplication canOpenURL:url]` will return true for all http/https links, because they can be opened with Safari, right? It will return false only for some completely unsupported schemes, like `qweasd`, correct? If yes, then what is the benefit of opening the same unsupported scheme in the else clause in `SafariViewController`?

Reply:
The benefit is to show the user that something went wrong and the link isn't supported without redirection into the other app. 
It's easier than create a custom alert.
If link is broken the sfsafarivc will open the link with the error code and description.

Reply:
Good, so the error is displayed in the VC. Thanks!